tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.flex

```diff
@@ -247,6 +248,60 @@ XML_COMMENT_CHAR = [^{}>\\-] | {XML_ESCAPED_SEQUENCE}
 XML_COMMENT_ALLOWED_SEQUENCE = {XML_BRACES_SEQUENCE} | {XML_COMMENT_SPECIAL_SEQUENCE} | ({XML_BRACES_SEQUENCE} {XML_COMMENT_SPECIAL_SEQUENCE})+ {XML_BRACES_SEQUENCE}? | ({XML_COMMENT_SPECIAL_SEQUENCE} {XML_BRACES_SEQUENCE})+ {XML_COMMENT_SPECIAL_SEQUENCE}?
 XML_COMMENT_SPECIAL_SEQUENCE = ">"+ | (">"* "-" ">"+)+ | "-"? ">"* "-"+
 
+
+// MARKDOWN_DOCUMENTATION
+MARKDOWN_DOCUMENTATION_LINE_START =  {HASH} {DOCUMENTATION_SPACE}?
+PARAMETER_DOCUMENTATION_START = {HASH} {DOCUMENTATION_SPACE}? {ADD} {DOCUMENTATION_SPACE}*
+RETURN_PARAMETER_DOCUMENTATION_START = {HASH} {DOCUMENTATION_SPACE}? {ADD} {DOCUMENTATION_SPACE}* {RETURN} {DOCUMENTATION_SPACE}* {SUB} {DOCUMENTATION_SPACE}*
+DOCUMENTATION_SPACE = [ ]
+
+// MARKDOWN_DOCUMENTATION_MODE
+DEFINITION_REFERERNCE = {REFERENCE_TYPE} {DOCUMENTATION_SPACE}+
+REFERENCE_TYPE = {TYPE}|{ENDPOINT}|{SERVICE}|{VARIABLE}|{VAR}|{ANNOTATION}|{MODULE}|{FUNCTION}|{PARAMETER}
+MARKDOWN_DOCUMENTATION_TEXT = {DOCUMENTATION_TEXT_CHARACTER}+
+DOCUMENTATION_TEXT_CHARACTER =  [^`\n+\-] | '\\' {BACKTICK}
+DOCUMENTATION_ESCAPED_CHARACTERS = {DOCUMENTATION_SPACE} | [+-]
+MARKDOWN_DOCUMENTATION_LINE_END = [\n]
+TYPE = "type"
+ENDPOINT = "endpoint"
+SERVICE = "service"
+VARIABLE = "variable"
+VAR = "var"
+ANNOTATION = "annotation"
+MODULE = "module"
+FUNCTION = "function"
+PARAMETER = "parameter"
+HASH = "#"
+ADD = "+"
+SUB = "-"
+RETURN = "return"
+
+// SINGLE_BACKTICKED_MARKDOWN
+SINGLE_BACKTICK_CONTENT = (([^`\n] | '\\' {BACKTICK})* [\n])? ({MARKDOWN_DOCUMENTATION_LINE_START} ([^`\n] | '\\' {BACKTICK})* [\n]?)+ | ([^`\n] | '\\' {BACKTICK})+
+SINGLE_BACKTICK_MARKDOWN_START = {BACKTICK}
+SINGLE_BACKTICK_MARKDOWN_END =  {BACKTICK}
+
+// DOUBLE_BACKTICKED_MARKDOWN
+DOUBLE_BACKTICK_CONTENT =   (([^`\n] | {BACKTICK} [^`])* [\n])? ({MARKDOWN_DOCUMENTATION_LINE_START} ([^`\n] |
+{BACKTICK} [^`])* [\n]?)+ | ([^`\n] | {BACKTICK} [^`])+
+
+DOUBLE_BACKTICK_MARKDOWN_START = {BACKTICK} {BACKTICK}
+DOUBLE_BACKTICK_MARKDOWN_END = {BACKTICK} {BACKTICK}
+
+// TRIPLE_BACKTICKED_MARKDOWN
+TRIPLE_BACKTICK_CONTENT = (([^`\n] | {BACKTICK} [^`] | {BACKTICK} {BACKTICK} [^`])* [\n])? ({MARKDOWN_DOCUMENTATION_LINE_START} ([^`\n] | {BACKTICK} [^`] | {BACKTICK} {BACKTICK} [^`])* [\n]?)+
+                          |   ([^`\n] | {BACKTICK} [^`] | {BACKTICK} {BACKTICK} [^`])+
```

Comment:
Extra space after `|`. Also can we add clarification parenthesis for long and unclear conditions (everywhere)?

Reply:
Removed the extra spaces